meteor/client/ui/Settings/SettingsMenu.tsx

```diff
@@ -0,0 +1,511 @@
+									<FontAwesomeIcon icon={faPlus} />
+								</button>
+								{t('Blueprints')}
+							</h2>
+							<hr className="vsubtle man" />
+							{this.props.blueprints.map((blueprint) => (
+								<SettingsMenuBlueprint key={unprotectString(blueprint._id)} blueprint={blueprint} />
+							))}
+						</React.Fragment>
+					)}
+					<h2 className="mhs">{t('Devices')}</h2>
+					<hr className="vsubtle man" />
+					{this.props.peripheralDevices
+						.filter((device) => {
+							return device.subType === PeripheralDeviceAPI.SUBTYPE_PROCESS
+						})
+						.map((device) => (
+							<SettingsMenuPeripheralDevice key={unprotectString(device._id)} device={device} />
+						))}
+					<h2 className="mhs">{t('Tools')}</h2>
+					<hr className="vsubtle man" />
+					{(!MeteorSettings.enableUserAccounts || this.props.superAdmin) && (
+						<React.Fragment>
+							<NavLink
+								activeClassName="selectable-selected"
+								className="settings-menu__settings-menu-item selectable clickable"
+								to="/settings/tools/system"
+							>
+								<h3>{t('Core System settings')}</h3>
+							</NavLink>
+							<NavLink
+								activeClassName="selectable-selected"
+								className="settings-menu__settings-menu-item selectable clickable"
+								to="/settings/tools/migration"
+							>
+								<h3>{t('Upgrade Database')}</h3>
+							</NavLink>
+							<NavLink
+								activeClassName="selectable-selected"
+								className="settings-menu__settings-menu-item selectable clickable"
+								to="/settings/tools/snapshots"
+							>
+								<h3>{t('Manage Snapshots')}</h3>
+							</NavLink>
+						</React.Fragment>
+					)}
+				</div>
+			)
+		}
+	}
+)
+
+interface SettingsCollapsibleGroupProps {
+	links: Array<{ label: string; subPath: string }>
+	basePath: string
+}
+
+function SettingsCollapsibleGroup({
+	links,
+	children,
+	basePath,
+}: React.PropsWithChildren<SettingsCollapsibleGroupProps>) {
+	const [expanded, setExpanded] = React.useState(false)
+
+	const toggleExpanded = React.useCallback((e: React.MouseEvent) => {
+		e.preventDefault()
+		e.stopPropagation()
+
+		setExpanded((old) => !old)
+	}, [])
+
+	const location = useLocation()
+	React.useEffect(() => {
+		// Auto-expand if the current location starts with the basePath
+		if (location.pathname.startsWith(basePath)) {
+			setExpanded(true)
+		}
+	}, [location.pathname, basePath])
+
+	return (
+		<>
+			<NavLink
+				activeClassName="selectable-selected"
+				className="settings-menu__settings-menu-item selectable clickable"
+				to={basePath}
+				onClick={toggleExpanded}
+			>
+				{children}
+			</NavLink>
+			{expanded
+				? links.map((link, i) => (
+						<NavLink
+							key={i}
+							activeClassName="selectable-selected"
+							className="settings-menu__settings-menu-item selectable clickable"
+							to={`${basePath}/${link.subPath}`}
+						>
+							<div className="selectable clickable">
```

Comment:
Is this div useful for anything? The parent is already "selectable clickable", so this doesn't seem to be neccessary.